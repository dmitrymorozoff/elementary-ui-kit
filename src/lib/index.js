import { Button } from "./components/forms/Button";
import { Link } from "./components/forms/Link";
import { Switch } from "./components/forms/Switch";
import { Input } from "./components/forms/Input";
import { Checkbox } from "./components/forms/Checkbox";
import { Card } from "./components/others/Card";
import { Label } from "./components/others/Label";
import { CardButtons } from "./components/others/Card/components/CardButtons";
import { CardContent } from "./components/others/Card/components/CardContent";
import { CardMedia } from "./components/others/Card/components/CardMedia";
import { Collapse } from "./components/others/Collapse";
import { CollapseContent } from "./components/others/Collapse/components/CollapseContent";
import { CollapseTitle } from "./components/others/Collapse/components/CollapseTitle";
import { Modal } from "./components/others/Modal";
import { ModalContent } from "./components/others/Modal/components/ModalContent";
import { ModalTitle } from "./components/others/Modal/components/ModalTitle";
import { ModalActions } from "./components/others/Modal/components/ModalActions";
import { Text } from "./components/others/Text";
import { Search } from "./components/icons/Search";
import { RightArrow } from "./components/icons/RightArrow";
import { Plus } from "./components/icons/Plus";
import { DownArrow } from "./components/icons/DownArrow";
import { UpArrow } from "./components/icons/UpArrow";
import { Checked } from "./components/icons/Checked";
import { Select } from "./components/forms/Select";
import { SelectItem } from "./components/forms/Select/components/SelectItem";

export {
    Button,
    Link,
    Switch,
    Input,
    Checkbox,
    Card,
    CardButtons,
    CardContent,
    CardMedia,
    Text,
    Search,
    RightArrow,
    DownArrow,
    UpArrow,
    Checked,
    Select,
    SelectItem,
    Label,
    Plus,
    Collapse,
    CollapseContent,
    CollapseTitle,
    Modal,
    ModalContent,
    ModalTitle,
    ModalActions,
};
