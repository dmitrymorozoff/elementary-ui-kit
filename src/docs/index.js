import React from "react";
import ReactDOM from "react-dom";
import { CollapseContent, Collapse, CollapseTitle } from "../lib";

const App = () => (
    <div>
        <Collapse>
            <CollapseTitle>Ornithorhynchus anatinus</CollapseTitle>
            <CollapseContent>
                The platypus (Ornithorhynchus anatinus), sometimes referred to as the duck-billed platypus, is a
                semiaquatic egg-laying mammal endemic to eastern Australia, including Tasmania.
                <br />
                <br />
                Together with the four species of echidna, it is one of the five extant species of monotremes, the only
                mammals that lay eggs instead of giving birth to live young. The animal is the sole living
                representative of its family (Ornithorhynchidae) and genus (Ornithorhynchus), though a number of related
                species appear in the fossil record.
                <br />
                <br />
                The first scientists to examine a preserved platypus body (in 1799) judged it a fake, made of several
                animals sewn together
            </CollapseContent>
        </Collapse>
    </div>
);

// eslint-disable-next-line
ReactDOM.render(<App />, document.getElementById("root"));
