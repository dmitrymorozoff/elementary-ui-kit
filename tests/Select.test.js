import React from "react";
import { Select, SelectItem } from "../src/lib";

describe("<Select/>", () => {
    it("should render a default select", () => {
        const component = shallow(
            <Select>
                <SelectItem value={1}>One</SelectItem>
                <SelectItem value={2}>Two</SelectItem>
                <SelectItem value={3}>Three</SelectItem>
            </Select>,
        );
        expect(component).toMatchSnapshot();
    });
});
