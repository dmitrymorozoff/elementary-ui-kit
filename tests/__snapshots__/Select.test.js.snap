// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select/> should render a default select 1`] = `
<styled.div
  fullWidth={false}
>
  <styled.div
    disabled={false}
    onClick={[Function]}
  >
    <styled.input
      color="#3a77f8"
      disabled={false}
      placeholder=""
      readOnly={true}
      value=""
    />
    <styled.div
      open={false}
    >
      <DownArrow
        fill="#abb3c8"
        size="14px"
        style={Object {}}
      />
    </styled.div>
  </styled.div>
  <styled.div
    color="#3a77f8"
    open={false}
  >
    <styled.div
      disabled={false}
      key="0"
      onClick={[Function]}
      value=""
    >
      One
    </styled.div>
    <styled.div
      disabled={false}
      key="1"
      onClick={[Function]}
      value=""
    >
      Two
    </styled.div>
    <styled.div
      disabled={false}
      key="2"
      onClick={[Function]}
      value=""
    >
      Three
    </styled.div>
  </styled.div>
</styled.div>
`;
